sklearn/utils/_array_api.py

```diff
@@ -109,16 +118,22 @@ def _check_array_api_dispatch(array_api_dispatch):
                 " the API specification"
             )
         if os.environ.get("SCIPY_ARRAY_API") != "1":
```

Comment:
Currently the source of truth is `scipy._lib._array_api.SCIPY_ARRAY_API` not necessarily the `SCIPY_ARRAY_API` environment variable which can be set too late for example after having imported scipy.

https://github.com/scipy/scipy/blob/9525f45c1cb9068bcb8e02d8fd81abc0c6f4e4ae/scipy/_lib/_array_api.py#L31-L32

Reply:
Done